### PR TITLE
*WIP* Add: cypress tests setup

### PR DESCRIPTION
**Pull request by MehulKChaudhari**
This PR will add end-to-end test and jest setup to the Repo.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,39 +1,45 @@
...
+  "settings": {
+    "react": {
+      "version": "detect"
     }
-  }
\ No newline at end of file
+  },
+  "rules": {
+    "linebreak-style": ["warn", "unix"],
+    "no-unused-vars": "warn",
```
Comment (harshith-venkatesh) [inline .eslintrc:33]: Should we not consider this as 'error' or 2?

File: package.json [hunk 2]
```diff
@@ -9,8 +9,11 @@
     "build": "next build",
     "start": "next start -p $PORT",
     "build-static": "next build && next export",
-    "test": "jest --coverage",
+    "cy:open": "cypress open",
+    "test:jest": "jest --coverage",
```
Comment (harshith-venkatesh) [inline package.json]: We could add --watchAll --detectOpenHandles --silent=false to ensure we see the errors for the tests created

Comment (harshith-venkatesh) [reply package.json]: Would it be better to create coverageThreshold?
https://jestjs.io/docs/configuration#coveragethreshold-object

File: .eslintrc (hunk 1 above)
Comment (MehulKChaudhari) [reply .eslintrc:33]: Sorry I didn't get you here.

**Comment by prakashchoudhary07** [comment]
@MehulKChaudhari excellent work on setting up MSW, but I couldn't find it as devDependencies or dependencies in package.json. 

NIT: Can we separate the MSW setup into other PR, Also lets move to yarn and start using volta to pin node versions

**Comment by MehulKChaudhari** [comment]
> @MehulKChaudhari excellent work on setting up MSW, but I couldn't find it as devDependencies or dependencies in package.json.
> 
> NIT: Can we separate the MSW setup into other PR, Also lets move to yarn and start using volta to pin node versions

Sure

File: jest.config.js [hunk 3]
```diff
@@ -5,6 +5,32 @@ module.exports = {
...
+
+    // Handle CSS imports (without CSS modules)
+    '^.+\\.(css|sass|scss)$': '<rootDir>/__mocks__/styleMock.js',
+
+    /* Handle image imports
+    https://jestjs.io/docs/webpack#handling-static-assets */
+    '^.+\\.(jpg|jpeg|png|gif|webp|svg)$': '<rootDir>/__mocks__/fileMock.js',
+
+    // Handle absolute imports
+    '^@/(.*)$': '<rootDir>/src/$1',
```
Comment (ivinayakg) [inline jest.config.js]: root directory does not have any src folder

**Review by ivinayakg — CHANGES_REQUESTED** [review]
absolute imports are not working

File: package.json [hunk 4]
```diff
@@ -9,8 +9,11 @@
     "build": "next build",
     "start": "next start -p $PORT",
     "build-static": "next build && next export",
-    "test": "jest --coverage",
+    "cy:open": "cypress open",
+    "test:unit": "jest --coverage __tests__/unit",
```
Comment (prakashchoudhary07) [inline package.json:13]: How are we going to find integration tests coverage?

Comment (MehulKChaudhari) [reply package.json:13]: We don't have integration testing yet. we will do e2e testing

**Review by prakashchoudhary07 — DISMISSED** [review]

**Review by shubham-y — APPROVED** [review]

**Pull request merged**